packages/common/src/ui/layout/tables/RemoteDataTable.tsx

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable react/jsx-key */
+import React, { ReactNode, useEffect, useState } from 'react';
+
+import {
+  Column,
+  ColumnInstance,
+  Row,
+  usePagination,
+  useSortBy,
+  useTable,
+} from 'react-table';
+import { QueryObserverResult } from 'react-query';
+import clsx from 'clsx';
+
+import {
+  Box,
+  CircularProgress,
+  Grid,
+  makeStyles,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  TableContainer,
+  TablePagination,
+  Table as MuiTable,
+} from '@material-ui/core';
+
+import { SortAsc, SortDesc } from '../../icons';
+import { DEFAULT_PAGE_SIZE } from '.';
+
+export interface QueryProps {
+  first: number;
+  offset: number;
+  sort?: string;
+  desc?: boolean;
+}
+
+export interface QueryResponse<T> {
+  data: T[];
+  totalLength: number;
+}
+
+const useStyles = makeStyles(theme => ({
+  container: { marginBottom: 100 },
+  bodyCell: {
+    padding: 9.5,
+  },
+  clickable: {
+    cursor: 'pointer',
+  },
+  headerCell: {
+    ...theme.typography.th,
+  },
+  loadingIndicator: { marginLeft: 'auto', marginRight: 'auto' },
+  loadingIndicatorContainer: { display: 'flex', marginTop: 50 },
+}));
+
+interface TableProps<T extends Record<string, unknown>> {
```

Comment:
oh, typescript... what a glorious pain you can be sometimes.
I really like ts and the ability here to define a table of a certain data type is nice - but the hoops that you have to jump through some times can end up a little clumsy. Here I'm just trying to add a generic to the column and data, and it insists that I extend `object` and then the linter doesn't like the use of `object`. And cannot use `React.FC` with generics, which is why the prop type and return type of `JSX.Element` instead